src/slurm_plugin/clustermgtd.py

```diff
@@ -904,6 +1030,111 @@ def _terminate_orphaned_instances(self, cluster_instances, ips_used_by_slurm):
+        """Increase the partition failure count and add failed nodes to the set of bootstrap failure nodes."""
+        self._increase_partitions_protected_failure_count(bootstrap_failure_nodes)
+        # Update bootstrap_failure_nodes to include all nodes types have been failed during bootstrap
+        self._bootstrap_failure_nodes_types |= get_nodes_type(bootstrap_failure_nodes)
+
+    def _get_online_nodes_types(self, healthy_nodes):
+        """Find nodes types which have been failed during bootstrap now become online."""
+        online_nodes = self._find_online_slurm_nodes(healthy_nodes)
+        online_nodes_types = get_nodes_type(online_nodes)
+
+        return online_nodes_types
+
+    @staticmethod
+    def _get_partitions_to_disable(nodename_to_slurm_nodes_map, bootstrap_failure_partitions, partitions):
+        """Get partitions not running jobs to disable."""
+        # Get partitions have jobs running
+        partitions_have_active_jobs = ClusterManager._filter_partitions_with_active_jobs(
+            nodename_to_slurm_nodes_map, partitions
+        )
+        # Only disable bootstrap failure partitions that not having active jobs running
+        partitions_to_disable = set(bootstrap_failure_partitions) - partitions_have_active_jobs
+        bootstrap_failure_partitions_have_jobs = set(bootstrap_failure_partitions) - partitions_to_disable
+        if bootstrap_failure_partitions_have_jobs:
+            log.info(
+                f"Bootstrap failure partitions {bootstrap_failure_partitions_have_jobs} currently have jobs running, "
+                "not disabling them."
+            )
+        return partitions_to_disable
+
+    def _is_protected_mode_enabled(self):
+        """When protected_failure_count is set to -1, disable protected mode."""
+        if self._config.protected_failure_count == -1:
+            return False
+        return True
+
+    def _handle_protected_mode_process(self, slurm_nodes, partitions, bootstrap_failure_nodes, healthy_nodes):
+        """Handle the process of entering protected mode."""
+        # Handle successfully launched nodes
+        nodename_to_slurm_nodes_map = {node.name: node for node in slurm_nodes}
+        self._handle_successfully_launched_nodes(healthy_nodes)
+
+        # Handle bootstrap failures nodes
+        if bootstrap_failure_nodes:
+            log.warning("Found the following bootstrap failure nodes: %s", print_with_count(bootstrap_failure_nodes))
+            self._handle_bootstrap_failure_nodes(bootstrap_failure_nodes)
+
+        # Enter protected mode
+        # We will put a partition into inactive state only if the partition satisfies the following:
+        # Partition is not INACTIVE
+        # Partition bootstrap failure count above threshold
+        # Partition does not have job running
+        logging.info(f"Partitions_protected_failure_count_map is {self._partitions_protected_failure_count_map}")
```

Comment:
let's use log and not logging

Reply:
also let's try to have meaningful log messaged for customers

Reply:
* Changed into log
* Change the log messages
* Currently the log will for each node if the node have bootstrap failure. `"Node bootstrap error: Node %s is currently in replacement and no backing instance` like we do for unhealthy nodes. In other places, they are log for a set of nodes or partitions